Sources/BraintreeCore/ConfigurationCache.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+
+class ConfigurationCache {
+        
+    // MARK: - Private Properties
+    
+    private let timeToLiveMinutes = 5.0
+    
+    // MARK: - Internal Properties
+    
+    /// Exposed for testing
+    let cacheInstance = NSCache<NSString, BTConfiguration>()
+
+    // MARK: - Singleton Setup
+    
+    static let shared = ConfigurationCache()
+    private init() { }
+    
+    // MARK: - Internal Methods
+    
+    func putInCache(authorization: String, configuration: BTConfiguration) throws {
+        let cacheKey = try createCacheKey(authorization)
+        cacheInstance.setObject(configuration, forKey: cacheKey)
+    }
+    
+    func getFromCache(authorization: String) throws -> BTConfiguration? {
```

Comment:
Since this method throws can we throw and error here instead of returning nil then make BTConfiguration non optional?

Reply:
Ah I see we are returning nil because at the callsite if we don't have a cached config we want to fetch a new config. This feels a little odd to me since this method is called `getFromCache` I would expect us to always have the ability to return a config at this point. 🤔

Reply:
Good callout. The "throw" is bubbled up from if the base64 encoding fails. These methods don't _have_ to throw, the throw is always handled by APIClient and never bubbles up to the merchant, but I thought it was more clear to have "throw" to mean an error occurred & `nil` to mean there's actually no object in the cache. We _could_ lump both cases into a `nil` response, but that felt more Obj-C-ey.

Maybe there's a better name than `getFromCache`? `checkInCache`?

Open to ideas for making this public API more clear!

Reply:
(Also added some docstrings)